### ISSUE TITLE
extension_attribute type not handled correctly

### ISSUE THREAD

**Issue by torhoehn** [bitExpert/phpstan-magento#248]
I passed an array of SalesChannelInterfaces to the setSalesChannelInterface function:
`        $stockExtensionAttribute->setSalesChannels([$salesChannelInterface]);`

So I receive the following error from `phpstan`:
```
Parameter #1 $salesChannels of method Magento\InventoryApi\Api\Data\StockExtensionInterface::setSalesChannels() expects Magento\InventorySalesApi\Api\Data\SalesChannelInterface, array<int,
         Magento\InventorySalesApi\Api\Data\SalesChannelInterface> given.
```

The `extension_attribute` is defined here:
```
    <extension_attributes for="Magento\InventoryApi\Api\Data\StockInterface">
        <attribute code="sales_channels" type="Magento\InventorySalesApi\Api\Data\SalesChannelInterface[]" />
    </extension_attributes>
```

And the interface `\Magento\InventoryApi\Api\Data\StockExtensionInterface` looks like this:
```
    /**
     * @param \Magento\InventorySalesApi\Api\Data\SalesChannelInterface[]|null $salesChannels
     * @return void
     */
    public function setSalesChannels(?array $salesChannels): void;
```

So maybe `phpstan` handles the type of the extension_attribute not correctly?

**Comment by shochdoerfer** [comment]
Thanks for reporting this. Will need to check. I assume that the array type in the xml file is not properly interpreted.

**Comment by torhoehn** [comment]
No problem, I just ignored that line. :)

**Comment by shochdoerfer** [comment]
@torhoehn looks like my assumption is correct https://github.com/bitExpert/phpstan-magento/blob/91032ba3777bf22df981c959d70f0fdbc391e515/src/bitExpert/PHPStan/Magento/Autoload/DataProvider/ExtensionAttributeDataProvider.php#L113

Not sure why this was done, need to investigate.

**Comment by torhoehn** [comment]
@shochdoerfer btw it's not mentioned here that an array is possible as type: https://developer.adobe.com/commerce/php/development/components/attributes/#declare-extension-attributes

Don't know if this information is useful...

**Comment by shochdoerfer** [comment]
@torhoehn even though the array type is undocumented, it is working for you, I guess? :)

**Comment by torhoehn** [comment]
@shochdoerfer That's true. It's possible to pass an array.

**Comment by shochdoerfer** [comment]
I haven't ignored the issue, I just need some more time to catch up with work and all other things. Give me some more time to properly fix this.

**Comment by torhoehn** [comment]
@shochdoerfer No worries :)

**Comment by shochdoerfer** [comment]
@torhoehn fixed with the latest 0.23.1 release (I think). Mind giving it a try? Worked for my isolated test case but maybe I missed something else.

**Comment by torhoehn** [comment]
@shochdoerfer Seems to be work fine now. Thank you!
